pkg/core/xds/merge.go

```diff
@@ -0,0 +1,112 @@
+package xds
+
+import (
+	"encoding/json"
+	"reflect"
+	"strings"
+
+	jsonpatch "github.com/evanphx/json-patch/v5"
+)
+
+// indicates that all slices that start with this prefix will be appended, not replaced
+const appendSlicesPrefix = "Append"
+
+func MergeConfs(confs []interface{}) (interface{}, error) {
+	if len(confs) == 0 {
+		return nil, nil
+	}
+
+	resultBytes := []byte{}
+	for _, conf := range confs {
+		confBytes, err := json.Marshal(conf)
+		if err != nil {
+			return nil, err
+		}
+		if len(resultBytes) == 0 {
+			resultBytes = confBytes
+			continue
+		}
+		resultBytes, err = jsonpatch.MergePatch(resultBytes, confBytes)
+		if err != nil {
+			return nil, err
+		}
+	}
+
+	result, err := newConf(reflect.TypeOf(confs[0]))
+	if err != nil {
+		return nil, err
+	}
+
+	if err := json.Unmarshal(resultBytes, result); err != nil {
+		return nil, err
+	}
+
+	clearAppendSlices(reflect.ValueOf(result)) // clear appendable slices, so we won't duplicate values of the last conf
+	for _, conf := range confs {
+		appendSlices(reflect.ValueOf(result), reflect.ValueOf(&conf).Elem()) // call .Elem() to unwrap interface{}
+	}
+
+	v := reflect.ValueOf(result).Elem().Interface()
+
+	return v, nil
+}
+
+func clearAppendSlices(val reflect.Value) {
+	if val.Kind() != reflect.Struct && val.Kind() != reflect.Pointer {
+		panic("expected struct or pointer to a struct")
+	}
+	strVal := val
+	if val.Kind() == reflect.Pointer {
+		strVal = val.Elem()
+	}
+	if strVal.Kind() != reflect.Struct {
```

Comment:
Should we check here struct and pointer or just struct is enough? Like a message below: `expected struct or pointer to a struct`

Reply:
added with a func that Ilya proposed